src/Streamly/Internal/Data/Fold/Types.hs

```diff
@@ -537,55 +669,79 @@ lchunksOf2 n (Fold step1 initial1 extract1) (Fold2 step2 inject2 extract2) =
 -- @
 {-# INLINE lsessionsOf #-}
 lsessionsOf :: MonadAsync m => Double -> Fold m a b -> Fold m b c -> Fold m a c
-lsessionsOf n (Fold step1 initial1 extract1) (Fold step2 initial2 extract2) =
-    Fold step' initial' extract'
+lsessionsOf n (Fold stepS initialS extractS) (Fold stepC initialC extractC) =
```

Comment:
This code has been completely changed. The extension of the previous implementation is very buggy.

Read `XXX` above.